test/e2e/framework/framework.go

```diff
@@ -315,15 +314,14 @@ func (f *Framework) WaitForFleetCondition(t *testing.T, flt *agonesv1.Fleet, con
 			f.LogEvents(t, log, gs.ObjectMeta.Namespace, &gs)
 		}
 
-		return err
+		return resultErr
```

Comment:
Is this the line that you think may have made the test flaky?

Reply:
So this is the line that would let the `WaitForFleetCondition` seem to pass.

The line `gsList, err := f.ListGameServersFromFleet(flt)` would have overwritten the `err` variable (and it would be strange for it to fail) -- so if there was a timeout on the `wait.PollImmediate`, it wouldn't be caught and would allow the whole process to flow through, which is bad.

Reply:
Sorry - to answer the original question: I don't think this affects flakiness per-se -- but it does mean that the root cause of why e2e tests where failing was often harder to track down, because only a test that relies on the expected state of this test would fail after the fact.

Reply:
Got it.